Add regex option to the settings.

AprilHickson opened the pull request:
### The Problem: 
I couldn't rate limit all items with the urls 

`get:/api/category/type/{id}`

without also capturing the traffic for

`get:/api/category/type/template`

### The Solution:

Allow an option for Regex filtering.


### Notes:

I have cloned the wiki from this repo to here and made some appropriate changes

https://github.com/AprilHickson/AspNetCoreRateLimit/wiki

cristipufu commented:
Hi @AprilHickson, nice work!

Can you please write some proper examples? In these appSettings, you use the regex for the verbs (which shouldn't really be the case).

AprilHickson commented:
Thanks @cristipufu! 

I have updated both my version of the wiki, as I don't think I can do a PR for the wiki 

https://github.com/AprilHickson/AspNetCoreRateLimit/wiki/ClientRateLimitMiddleware#defining-rate-limit-rules 
and 
https://github.com/AprilHickson/AspNetCoreRateLimit/wiki/IpRateLimitMiddleware#defining-rate-limit-rules

And I have also updated the `appsettings.Regex.json` with some better examples of regex patterns.

Let me know if there are any other issues or if I had misunderstood your comment!
